From: Faidon Liambotis <paravoid@debian.org>
Date: Sat, 29 Aug 2015 17:05:01 +0200
Subject: udev: increase udev event timeout to 180s

Some kernel modules, like mptsas, can take longer then 30s to load so
udevd kills the (hanging) worker responsible for loading the module.

Increase timeout from 30s to 180s to workaround this issue, following
the upstream change which first increased the timeout from 30s to 60s
(2e92633) and subsequently to 180s (b5338a1).

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=787191
---
 src/udev/udevd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index a45d324..072499c 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -1415,7 +1415,7 @@ int main(int argc, char *argv[])
                                 if (worker->state != WORKER_RUNNING)
                                         continue;
 
-                                if ((now(CLOCK_MONOTONIC) - worker->event_start_usec) > 30 * USEC_PER_SEC) {
+                                if ((now(CLOCK_MONOTONIC) - worker->event_start_usec) > 180 * USEC_PER_SEC) {
                                         log_error("worker [%u] %s timeout; kill it", worker->pid,
                                             worker->event ? worker->event->devpath : "<idle>");
                                         kill(worker->pid, SIGKILL);
